### ISSUE TITLE
(aws-ec2): configuration so the default security group will not allow inbound and outbound traffic

### ISSUE THREAD

**Issue by SamStephens** [aws/aws-cdk#19394]
### Description

[AWS Foundational Security Best Practices](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html) state [The VPC default security group should not allow inbound and outbound traffic](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-ec2-2).

However the CDK does not provide any hooks to do this, or even any access to the default security group at all, other than [its ID via Vpc#vpcDefaultSecurityGroup](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#vpcdefaultsecuritygroup).

Lack of proper support for this in the CDK makes it difficult to meet the standards AWS is setting with the Foundational Security Best Practices. Looking at discussions in other issues, people are [creating custom CDK resources to perform this lockdown](https://github.com/aws/aws-cdk/issues/1606#issuecomment-1007358666).

### Use Case

To comply with security best practices as provided by [AWS Foundational Security Best Practices](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html).

### Proposed Solution

A boolean property on the VPC object, that defaults to `false` which keeps the existing behavior, and if set to true, will lock down the security group.

### Other information

Related issues:
* https://github.com/aws/aws-cdk/issues/4241
* https://github.com/aws/aws-cdk/issues/1606
* https://github.com/aws/aws-cdk/issues/1995

### Acknowledge

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by albertocorrales** [comment]
I was looking for something like this too, so I'm looking forward to having this implemented.

In the meantime, is there any way to workaround this? Like a command to get the default SG for a VPC and update the rules.

Thanks,

Alberto.

**Comment by SamStephens** [comment]
> I was looking for something like this too, so I'm looking forward to having this implemented.
> 
> In the meantime, is there any way to workaround this? Like a command to get the default SG for a VPC and update the rules.
> 
> Thanks,
> 
> Alberto.

The only workaround within the CDK I've seen is [using custom resources](https://github.com/aws/aws-cdk/issues/1606#issuecomment-1007358666) as I linked to in the issue description.

Outside of CDK, one interesting possibility is that AWS provides an example of [how to use AWS Config to detect and remediate this issue](https://github.com/aws-samples/aws-config-pci-fsbp-ssmremediations/blob/18ea8863525a4825b76dbd81d9f66b7f39bb24cc/aws-servicecatalog-prescriptivecompliance/cft/aws-servicecatalog-configremediations-v2.yml#L62-L105).

**Comment by SamStephens** [comment]
Thanks @corymhall

**Comment by mwebber** [comment]
@corymhall This is great, but it looks like the fix for this might be incomplete.

I upgraded to CDK 2.79.1 (which includes the fix), set the feature flag `"@aws-cdk/aws-ec2:restrictDefaultSecurityGroup": true`, and redeployed a Stack which defines a VPC.

I noticed that in Outbound Rules for the `default` security group, the rule allowing outbound IPv4 traffic had been removed, but the rule allowing outbound IPv6 traffic was still there (all IPv6 traffic allowed to `::/0`).
All Inbound Rules were removed as expected.

I observed this same result in all the VPCs that I re-deployed with the new flag enabled.
I did not test the initial deploy of a new VPC.
